pycloudlib/oracle/cloud.py

```diff
@@ -0,0 +1,218 @@
+# This file is part of pycloudlib. See LICENSE file for license information.
+"""OCI Cloud type."""
+
+import re
+import oci
+
+from pycloudlib.cloud import BaseCloud
+from pycloudlib.oracle.instance import OciInstance
+from pycloudlib.oracle.utils import wait_till_ready
+from pycloudlib.util import ubuntu_name_to_number
+
+
+class OCI(BaseCloud):
+    """OCI (Oracle) cloud class."""
+
+    _type = 'oci'
+
+    def __init__(self, tag, compartment_id, config_path='~/.oci/config'):
+        """
+        Initialize the connection to OCI.
+
+        OCI must be initialized on the CLI first:
+        https://github.com/cloud-init/qa-scripts/blob/master/doc/launching-oracle.md
+
+        Args:
+            tag: Name of instance
+            compartment_id: A compartment found at
+                https://console.us-phoenix-1.oraclecloud.com/a/identity/compartments
+            config_path: Path of OCI config file
+        """
+        super(OCI, self).__init__(tag)
+        self.compartment_id = compartment_id
+        self.config_path = config_path
+        config = oci.config.from_file(str(config_path))
+
+        self._log.debug('Logging into OCI')
+        self.compute_client = oci.core.ComputeClient(config)
+        self.network_client = oci.core.VirtualNetworkClient(config)
+
+    def delete_image(self, image_id):
+        """Delete an image.
+
+        Args:
+            image_id: string, id of the image to delete
+        """
+        self.compute_client.delete_image(image_id)
+
+    def released_image(self, release, operating_system='Canonical Ubuntu'):
```

Comment:
Do you have current use-cases in mind to have call-sites currently where we are supporting non-canonical Ubuntu?
Are you planning for centOS integration testing for cloud-init specifically?

Reply:
No, I initially hardcoded things to be Ubuntu, but thought for a library that might be used outside of just the cloud-init project, we might need greater flexibility. Do you have a concern with doing it this way, or just curious?

Reply:
This was only me being curious, thanks. I wasn't sure if you already has another specific consumer in mind.